RequestManagementSystem/private/RequestTask.py

```diff
@@ -80,7 +81,10 @@ def __init__( self, requestJSON, handlersDict, csPath, agentName, standalone = F
     gMonitor.registerActivity( "RequestOK", "Requests done",
                                "RequestExecutingAgent", "Requests/min", gMonitor.OP_SUM )
 
-    self.requestClient = ReqClient()
+    if requestClient is None:
```

Comment:
Ah, so that you can pass your mock for the test ?? Sorry but that's not really clean....